src/FS_plotting.jl

```diff
@@ -0,0 +1,303 @@
+
+
+
+
+#####PLOTTING THE FERMI SURFACE
+
+
+function plotFermiSurface2D(
+            k_values::Array{Float64, 2},
+            brillouin_zone:: BZ,
+            kappa::Float64;
+            plot_title::String="",
+            plot_color="b",
+            figsize::Tuple=(6,6),
+            showPlot::Bool=true,
+            save_filename::String="NONE"
+        )   where {L,NB, BZ<:AbstractBrillouinZone{R} where {
+                 R <: AbstractReciprocalUnitcell{P,B} where {
+                 P <: AbstractReciprocalPoint{D} where {D},
+                 B<:AbstractBond{L,NB},
+}}}
+
+
+    ###########################
+    #   INITIAL SETTINGS
+    ###########################
+
+    # configure plot environment
+    rc("font", family="serif")
+
+    # create a new figure
+    fig = figure(figsize=figsize)
+
+
+
+    ###########################
+    #   PLOT FERMI SURFACE
+    ###########################
+
+    # if brillouin zone not empty, plot it as well
+     if length(corners(brillouin_zone)) > 0
+        plotBrillouinZone(brillouin_zone, new_figure=false)
+    end
+
+    # scatter the points
+    
+    scatter(k_values[:,1], k_values[:,2], color=plot_color)
+
+
+
+    ###########################
+    #   CONFIGURE AXIS & TITLE
+    ###########################
+
+    # set the title
+    if plot_title == "AUTO"
+        # set the title to an automatically generated title
+        title("Fermi surface")
+    elseif plot_title == ""
+        title("κ=$(kappa)")
+    else
+        # set the title to the given title
+        title(plot_title)
+    end
+
+    # equal axis
+    gca()[:set_aspect]("equal")
```

Comment:
Note that this will produce a deprecation warning (or won't work at all) for newer PyPlot.jl versions as `obs[:method]()` has been deprecated in favor of the simpler `obs.method()`. 

There are many more instances below.

Reply:
I noticed and left it like this because I did not have time to review the code in detail - but thanks for the note!